---
title: "Benchmarking figures"
output: pdf_document
---

```{r,settings_and_lib,echo=F,message=F,}
require(ggplot2)
require("plyr")
require("Biostrings")
require(reshape2)
require("pROC")
require(gridExtra)
require(cowplot)
require(gtable)
require(png)
theme_set(new = theme_classic())
require(knitr)
opts_chunk$set(fig.align = "center", warning = FALSE, echo = FALSE,message=F,cache=T)

rhg_cols<-c("#08519cff","#3182b4ff","#6baed6ff","#b1bdc5ff","#787878ff")

dd=4 # digits in tables
```
 
### Figure 1
```{r, roc_skematic,fig.height=4,fig.width=3.4}
set.seed(42) # make the figures reproducible
example<-data.frame(Condition=c(rep(x = c(TRUE,FALSE),each=20)),test=c(runif(20,-10,-4),runif(20,-6,0))) # Make a dataframe of T and F positives with random p values that favors a lower distribution for the TP
ex.roc<-roc(example$Condition~example$test) # make an ROC based on this data

cord<-coords(ex.roc,x="all") # get the coordinates of the ROC
cord.l<-melt(cord) # melt the coodinates for ease of plotting
cord.l<- mutate(cord.l,Id=c(0,head(as.numeric(rownames(cord.l))%/%3,-1))) # Add an Id so that the specificity and sensitivity at a given threshold can be groupled together

# Set an id column to group the threshold,specificity, and sensitivity together with the same number by integer division. I adjust with c(0, head ..., -1) since 3%/%3=1 but it should be grouped with the 0's.
roc.df<-dcast(cord.l,Id~Var1) # dcast into columns of threshold, specificity and sensitivity
roc.df<-roc.df[order(roc.df$sensitivity),] # So that the points connect nicely

#Helper function that gets the threshold and sensitivity from and roc at a given speceificity
helper.roc<-function(spec,roc.df){
  above.sense.cut<-subset(roc.df,specificity>=spec,select=c(threshold,specificity,sensitivity)) # get the points where specificity is above the cutoff
  sense.min<-min(above.sense.cut$specificity) # get the minimum of these to get as close to desired sensitivity as possible
  sense.cut<-subset(above.sense.cut,specificity==sense.min) # get all rows where sensitivity is at this point
  spec.max<-max(sense.cut$sensitivity) # but there can be different FPR for each TPR so here we maximize specificity
  output<-subset(sense.cut,sensitivity==spec.max)
}

plot.ex<-function(spec1,spec2,roc.df,distribution){ 
  cut1<-helper.roc(spec1,roc.df)#dataframe with the threshold,sensitivity and specificity at the given specificity
  cut2<-helper.roc(spec2,roc.df)
  output<-rbind(cut1,cut2)
  lines<-data.frame(sense=c(output$sensitivity,output$sensitivity,0,0), # make a data frame with the coordinates for the lines whicha re to be added
                    spec=c(0,0,1-output$specificity,1-output$specificity), # really 1-specificity
                    group=rep(c(spec1,spec2),times=3))
  
  #plot the roc
  roc.p<-ggplot(roc.df,aes(x=1-specificity,y=sensitivity))+geom_step(color="black",size=1.5)+xlab("1-Specificity")+ylab("Sensitivity")+theme(axis.title.y=element_text(vjust=1))
  roc.p<-roc.p+geom_line(data=lines,aes(x=spec,y=sense,group=as.factor(group),color=rep(c("#08519cff","#787878ff"),times=3)),size=c(1),lty=c(2))+scale_color_manual(values=c("#08519cff","#787878ff"))
    
 roc.p<-roc.p+theme(axis.title.y = element_text(vjust=1.2))+theme(axis.title.x = element_text(vjust=0))+theme(legend.position="none")
  
 
 #plot the points and cut offs
 points.p<-ggplot(distribution,aes(x=1,y=test,fill=Condition))+geom_dotplot(binaxis = "y", stackdir = "center",dotsize = 1.5)+scale_fill_manual(values=c("white","black"),name="",breaks=c(T,F),labels=c("True Positive","False Positive"))+xlab("")+ylab("Log(p value)")+theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(),axis.title.y = element_text(vjust=1.2),legend.position=c(0.9,0.65),legend.background = element_rect(fill = alpha("white",0)))+xlab("")
 
 points.p<-points.p+geom_hline(yintercept=output$threshold,
                          size=1,lty=2,color=c("#08519cff","#787878ff")) # these are the first 2 colors in set 1
  
ggdraw() +
  draw_plot(points.p, 0, .45, .8, 0.48) + #x y width height
  draw_plot(roc.p, 0, 0, .8, 0.48) +
  draw_plot_label(c("A", "B"), c(0.05, 0.05), c(1, 0.55), size = 15)

 
 
}
plot.ex(0.35,0.95,roc.df,example)

pdf("../results/figures/roc.example.pdf",width=3.4,height=4)
plot.ex(0.35,0.95,roc.df,example)
dev.off()
pdf("~/Box Sync/Lab_work/Talks/2016_812/figures/roc.ex.pdf",width = 12,height = 8)
plot.ex(0.35,0.95,roc.df,example)
dev.off()

```


Figure 1. An example ROC curve. A) Hypothetical variants are stratified by log of the p value. True positives variants are shown as closed circles, while false positive variants are represented as open circles. Potential thresholds are indicated as colored lines. B) An ROC curve made from the hypothetical data shown in A. The colored lines indicate the points on the curve made by the thresholds in A. 
\pagebreak
```{r,roc_helper_runctions}
find_freq<-function(Id){ # helper function to get the expected frequency from the sample name
  if(Id=="Covaris_5"){
    x=0.05} else if ( Id=="Covaris_25"){
    x=0.025} else if ( Id=="Covaris_125"){
    x=0.0125} else if ( Id=="Covaris_063"){
    x=0.0063} else if ( Id=="Covaris_016"){
    x=0.0016}
}

find_freq.v<-Vectorize(find_freq) # vectorized version of the helper function above


find_freq.h<-function(Id){ # helper function to get the expected frequency from the sample name
  if(Id==0.05){
    x="5.0%"} else if ( Id==0.02){
    x="2.0%"} else if ( Id==0.01){
    x="1.0%"} else if ( Id==0.005){
    x="0.5%"} else if ( Id==0.002){
    x="0.2%"}
}

find_freq.v.h<-Vectorize(find_freq.h) # vectorized version of the helper function above





id<-function(x){
  x<-strsplit(x,".",fixed=T)[[1]][1]
}

seg<-function (x) {
  x <- strsplit(x, ".", fixed = T)[[1]][2]
}  

fill_in_plots<-function(x){
  spec<-min(x$adj.specificity[which(x$adj.specificity>0.995)])
  sense<-max(x$adj.sensitivity[x$adj.specificity==spec]) # incase there is a step right here
  
  y<-data.frame(threshold=x$threshold[1],Id=x$Id[1],adj.specificity=0.9950001,adj.sensitivity=sense,samp=unique(x$samp),exp.freq=x$exp.freq[1],FP=x$FP[1],TP=x$TP[1],sensitivity=x$sensitivity[1],specificity=x$specificity[1])
  
  if("gc" %in% names(x)){
    y$gc=x$gc[1]
  }
  return(y)
}
sum_roc<-function(x,direction){
  sum.df<-subset(x,category %in% c(TRUE,FALSE),select=c(category,p.val))# get the TRUE and false variant calls and p.vals
  if(length(which(sum.df$category==T))>0){ # filter out cases where there aren't any TP found
    if(length(which(sum.df$category==F))==0){
      sum.df<-rbind(sum.df,data.frame(category=F,p.val=1))
    }  
    roc(sum.df$category~sum.df$p.val,plot=F,CI=T,direction=direction)
  }
}

roc_df<-function(roc_analysis){ # get the coordinates and cut offs for all the points in the ROC object
  roc_analysis<-roc_analysis[unlist(lapply(roc_analysis,is.null))==F]
  all<-lapply(roc_analysis,coords,x="all")
  all.long<-lapply(all, melt) 
  all.long<-lapply(all.long, function(x){ 
    mutate(x, 
           Id=c(0,head(as.numeric(rownames(x))%/%3,-1))
    )
  }) # Set an id column to group the threshold,specificity, and sensitivity together with the same number by integer division. I adjust with c(0, head ..., -1) since 3%/%3=1 but it should be grouped with the 0's.
  roc.ls<-lapply(all.long,function(x) dcast(x,Id~Var1)) # dcast into columns of threshold, specificity and sensitivity
  roc.df<-do.call(rbind,roc.ls) # combine into data frame
  roc.df$samp<-unlist(lapply(rownames(roc.df),id)) # add sample id column
  
  mutate(roc.df, exp.freq=find_freq.v(samp))-> roc.df # this yeilds null in the hiseq.roc function below
}  

adjust.coords<-function(roc.df,sum.df,possible_tp,possible_vars){ # adjust the sensitivity and specificity called by pROC  
  samp<-roc.df$samp[1]
  samp.df<-subset(sum.df,Id==samp)
  sense.factor<-length(which(samp.df$category==T))/possible_tp
  TN.samp<-length(which(samp.df$category==F))
  mutate(roc.df,adj.sensitivity=sensitivity*sense.factor,FP=(TN.samp-TN.samp*specificity),TP=adj.sensitivity*possible_tp,adj.specificity=(possible_vars-possible_tp-FP)/((possible_vars-possible_tp)))
}

roc_df.one<-function(roc_anal,thr){ # get the coordinates and cut offs for all the points in the ROC object
  roc_analysis<-roc_anal[unlist(lapply(roc_anal,is.null))==F]
  all<-lapply(roc_analysis,coords,x=thr,input="thr")

  #roc.ls<-lapply(all.long,function(x) dcast(x,Id~Var1)) # dcast into columns of threshold, specificity and sensitivity
  roc.df<-as.data.frame(do.call(rbind,all)) # combine into data frame
  roc.df$samp<-rownames(roc.df) # add sample id column
  
  mutate(roc.df, exp.freq=find_freq.v(samp))-> roc.df
} 
######### Miseq roc functions ##########
miseq.roc<-function(sum.df,possible_tp,possible_vars,direction){ # A function to run the roc calculations and adjustments
  roc.ls<-dlply(sum.df,~Id,sum_roc,direction)

  roc.df<-roc_df(roc.ls)
  

  roc.df.adj<-ddply(roc.df,~samp,adjust.coords,sum.df,possible_tp,possible_vars)
  
  roc.df.adj<-mutate(roc.df.adj,exp.freq=as.factor(find_freq.v(samp)))
  #fills<-ddply(roc.df.adj,~samp,fill_in_plots) # take out these lines to not fill in the plots
  #roc.df.adj<-rbind(fills,roc.df.adj) # here too
  roc.df.adj<-roc.df.adj[order(roc.df.adj$adj.sensitivity),]
  roc.df.adj$exp.freq <- factor(roc.df.adj$exp.freq, levels = rev(levels(roc.df.adj$exp.freq)))
  return(roc.df.adj)
}
miseq.roc.table<-function(sum.df,cut.off,possible_tp,possible_vars,direction){
  roc.ls<-dlply(sum.df,~Id,sum_roc,direction)
  roc.df<-roc_df.one(roc.ls,cut.off)
  roc.df.adj<-ddply(roc.df,~samp,adjust.coords,sum.df,possible_tp,possible_vars)
  roc.df.adj<-roc.df.adj[order(roc.df.adj$exp.freq,decreasing = T),]
  roc.table<-subset(roc.df.adj,select=c(exp.freq,adj.sensitivity,TP,adj.specificity,FP))
  return(roc.table)
}

# The ROC data only includes points there the sensitivity changes.  At this point I have limitted our analysis to variants with p<0.05. So the lines in some plots stop short. For the final draft I will not apply this cut off. Even in that future case this step may be need as the ROC data only includes points where the sensitivity changes.  For now   This function extents the lines by adding a data point at the specificity of 0.002 and appropriate sensitivity. At the most it hurts us as we do not increase sensitivity and it might , but I doubt it. Anyway it is of no concern as we only care about the p=0.01 case which is well contained before this extension is needed

######### Hiseq roc functions ##########
hiseq.roc<-function(sum.df,possible_tp,possible_vars,direction){ # A function to run the roc calculations and adjustments
  roc.ls<-dlply(sum.df,~Id,sum_roc,direction)

  roc.df<-roc_df(roc.ls)


  roc.df.adj<-ddply(roc.df,~samp,adjust.coords,sum.df,possible_tp,possible_vars)
  roc.df.adj<-roc.df.adj[order(roc.df.adj$adj.sensitivity),]
  mutate(roc.df.adj, gc = as.numeric(sub(pat,"\\1",samp)), # This fills in the null column given in the roc_df function above. line 161
       exp.freq = sub(pat,"\\2",samp)
  )-> roc.df.adj
  fills<-ddply(roc.df.adj,~samp,fill_in_plots) # take out these lines to not fill in the plots
  roc.df.adj<-rbind(fills,roc.df.adj) # here too
  roc.df.adj$exp.freq<-as.character(roc.df.adj$exp.freq) # so that teh calculations below work since it is a factor here
  mutate(roc.df.adj,exp.freq=ifelse(test = grepl("0",exp.freq),yes=as.numeric(exp.freq)/1000,no = as.numeric(exp.freq)/100))->roc.df.adj
  roc.df.adj$exp.freq<-as.factor(roc.df.adj$exp.freq)
  roc.df.adj$exp.freq <- factor(roc.df.adj$exp.freq, levels = rev(levels(roc.df.adj$exp.freq)))
  return(roc.df.adj)
}

hiseq.roc.table<-function(sum.df,cut.off,possible_tp,possible_vars,direction){
  roc.ls<-dlply(sum.df,~Id,sum_roc,direction)
  roc.df<-roc_df.one(roc.ls,cut.off)
  roc.df.adj<-ddply(roc.df,~samp,adjust.coords,sum.df,possible_tp,possible_vars)
mutate(roc.df.adj, gc = as.numeric(sub(pat,"\\1",samp)),
       exp.freq = sub(pat,"\\2",samp)
)-> roc.df.adj

mutate(roc.df.adj,exp.freq=ifelse(test = grepl("0",exp.freq),yes=as.numeric(exp.freq)/1000,no = as.numeric(exp.freq)/100))->roc.df.adj
roc.df.adj<-roc.df.adj[order(roc.df.adj$exp.freq,decreasing = T),]
roc.table<-subset(roc.df.adj,select=c(gc,exp.freq,adj.sensitivity,TP,adj.specificity,FP))
return(roc.table)
}



### tableGrob functions and settings #####
tt1<-ttheme_minimal(colhead=list(fg_params=list(fontface=1,cex=0.75)),core=list(fg_params=list(cex=0.75)))#,fontfamily="serif")))
make.table<-function(df,theme){
  tbl<-tableGrob(df,rows=NULL,theme=theme) # Make table
  tbl$widths <- unit(rep(1/ncol(tbl), ncol(tbl)), "npc")
  tbl<-gtable_add_grob(tbl,
        grobs = grobTree(
          segmentsGrob( # line across the bottom
            x0 = unit(0,"npc"),
            y0 = unit(0,"npc"),
            x1 = unit(1,"npc"),
            y1 = unit(0,"npc"),
            gp = gpar(lwd = 2.0)),
        segmentsGrob( # line across the top
            x0 = unit(0,"npc"),
            y0 = unit(1,"npc"),
            x1 = unit(1,"npc"),
            y1 = unit(1,"npc"),
            gp = gpar(lwd = 2.0))),
        t = 1, b = 1, l = 1, r = ncol(tbl))

  tbl<-gtable_add_grob(tbl,
                     grobs=segmentsGrob(
                       x0 = unit(0,"npc"),
            y0 = unit(0,"npc"),
            x1 = unit(1,"npc"),
            y1 = unit(0,"npc"),
            gp = gpar(lwd = 2.0)),
        t = nrow(tbl), b = nrow(tbl), l = 1, r = ncol(tbl))
}
```


### Figure 2
```{r,Miseq_out_of_the_box,fig.width=3.4,fig.height=7}

sum.df<-read.csv("../data/process/2014-5-30/Variants/all.sum.csv",stringsAsFactors=F,comment.char = '#')
  
true_snv<-read.csv("../data/reference/PR8_WSN33.csv",comment.char = '#') # read in a csv of the differences in PR8 and WSN33 as determined by sanger
true_snv<-subset(true_snv,Ref!="-" & Allele.1!="-") # remove any indels as we are not concerned with them here
mutate(true_snv,mutant=paste0(Name,"_",Ref,Ref.Pos,Allele.1))->true_snv #make a column that has names the variants as segment_ReferencePositionVariant
true_snv<-rename(true_snv,c("Name"="chr","Ref.Pos"="pos")) # rename the columns for use later
PR8_var<-read.csv("../data/reference/PR8_variants.csv",comment.char = "#") # read in differences between plasmid control on viral PR8 used in dilutions


mutate(PR8_var,mutant=paste0(Name,"_",Ref,Ref.Pos,Allele.1))->PR8_var #make a column that has names the variants as segment_ReferencePositionVariant
PR8_var<-rename(PR8_var,c("Name"="chr","Ref.Pos"="pos")) # rename the columns for use later  
sum.df<-mutate(sum.df,category=mutation %in% true_snv$mutant) # add Column for True and false variants to variant calls
sum.df$category[sum.df$mutation %in% PR8_var$mutant]<-"PR8" # id the variants found in PR8
sum.df<-mutate(sum.df,exp.freq=find_freq.v(Id)) # add the expected frequency of the true variants


# Get information about the regions we can investigate based on the primer sites


regions.bed <-read.csv("../data/reference/pr8_noprimingsites.bed.csv",stringsAsFactors = F,comment.char = "#")
regions.bed<-mutate(regions.bed,length=stop-start-1) #because the numbers in the csv are the inner most binding sites of the primers and represent the last positiions that cannot be interogated.

possible_vars<-sum(regions.bed$length)*3 # three possible variants/ position
miseq_vars<-sum(regions.bed$length)*3
miseq_kb<-sum(regions.bed$length)/1000

# limit the PR8 and the true positive data.frames to those not excluded by the priming sites

primer.cut<-function(x){ # a helper function to do this
  chr<-unique(x$chr)
  start<-regions.bed$start[match(x$chr,regions.bed$chr)]
  stop<-regions.bed$stop[match(x$chr,regions.bed$chr)]
  
  subset(x,pos>start & pos<stop)
}
sum.df<-ddply(sum.df,~chr,primer.cut) # interogating only the sites within primer location removes 244 called variants for two sided and 406 for one sided test
true_snv<-ddply(true_snv,~chr,primer.cut) # interogating only the sites within primer location removes 8 potential true positive (491)
PR8_var<-ddply(PR8_var,~chr,primer.cut) # interogating only the sites within primer location removes 12 potential PR8 differences (15)



## These functions make an ROC by allowing pROC to calculate the positions based on only the variants that are provided, and then we adjust the senesitivity for with what we know.  For example if only 3 TP are found in one sample pROC will call that 100% Sensitivity but we will adjust it to 3/possible_tp and so forth.

tp<-unique(true_snv$mutant)
possible_tp<-length(tp)


m.roc.df<-miseq.roc(sum.df,possible_tp,possible_vars,">")

miseq.roc.p<-ggplot(m.roc.df,aes(x=1-adj.specificity,y=adj.sensitivity))+geom_step(aes(color=exp.freq),size=2)+scale_color_manual(values=rhg_cols,name="Frequency",labels=c("5.0%","2.5%","1.25%","0.63%","0.016%"))+xlab(bquote("1-Specificity (x"~10^-3~")"))+ylab("Sensitivity")+scale_y_continuous(limits=c(0,1))+scale_x_continuous(limits=c(0,0.005),breaks=c(0,0.001,0.002,0.003,0.004,0.005),labels=c(0:5))+theme(legend.position="none")

#miseq.roc.p




# pdf("~/Box Sync/Lab_work/Talks/2016_812/figures/miseq_roc.one.sided.pdf")
#  miseq.roc.p
# dev.off()

m.roc.table<-miseq.roc.table(sum.df,0.01,possible_tp,possible_vars,">")


#knitr::kable(m.roc.table)
m.roc.table<-rename(m.roc.table,c("exp.freq"="Frequency","adj.sensitivity"="Sensitivity","TP"="True\nPositives","adj.specificity"="Specificity","FP"="False\nPositives"))

m.roc.table$Frequency<-c("5.0%","2.5%","1.25%","0.63%","0.16%")
m.roc.table$Sensitivity<-round(m.roc.table$Sensitivity,digits=dd)
m.roc.table$Specificity<-round(m.roc.table$Specificity,digits=dd)

miseq.sense.63<-format(m.roc.table$Sensitivity[m.roc.table$Frequency=="0.63%"],digits = dd)
miseq.sense.16<-m.roc.table$Sensitivity[m.roc.table$Frequency=="0.16%"]

tbl<-make.table(m.roc.table,tt1)


readPNG("../results/pictures/miseq_setup.png")->img # read in set up png
g <- rasterGrob(img, interpolate=TRUE)
qplot(1:1, 1:1, geom="blank") +
annotation_custom(g, xmin=-Inf, xmax=Inf, ymin=-Inf, ymax=Inf) +
geom_blank()+xlab("")+ylab("")+
theme(line = element_blank(),
text = element_blank(),
line = element_blank(),
title = element_blank())->miseq.setup # empty plot with setup png a background



#plot_grid(miseq.setup,miseq.roc.p, tbl, labels=c("A", "B","C"), ncol = 3, nrow = 1)
# Figure 2

pdf("../results/figures/miseq1.pdf",width=3.4,height=7)
ggdraw() +
  draw_plot(miseq.setup, 0.2, .66, .6, 0.33) + #x y width height
  draw_plot(miseq.roc.p, 0.05, 0.33, .9, 0.33) +
  draw_plot(tbl, 0.05, 0, .9, 0.33) +
  draw_plot_label(c("A", "B", "C"), c(0.01, 0.01, 0.01), c(0.95, 0.7, 0.35), size = 15)
dev.off()

ggdraw() +
  draw_plot(miseq.setup, 0.2, .66, .6, 0.33) + #x y width height
  draw_plot(miseq.roc.p, 0.05, 0.33, .9, 0.33) +
  draw_plot(tbl, 0.05, 0, .9, 0.33) +
  draw_plot_label(c("A", "B", "C"), c(0.01, 0.01, 0.01), c(0.95, 0.7, 0.35), size = 15)


#ggplot(subset(sum.df,p.val<0.01),aes(x=pos,y=Read_pos,color=category))+geom_point()+facet_wrap(~chr)+geom_hline(yintercept=c(63,188))


#ggplot(sum.df,aes(x=Read_pos,fill=category))+geom_histogram(position="dodge")





```


Figure 2. Initial DeepSNV accuracy. A) Reconstitued cDNA genomes of influenza strain WSN33 were diluted serially into PR8 prior to sequencing on an Illumina Miseq machine. B) An ROC curve measuring the accuracy of DeepSNV in identifying WSN33 variants mixed with PR8 at the indicated frequencies. C) A summary the data in B at a p value threshold of 0.01.
\pagebreak

```{r,Miseq_read.cut,eval=F}
### With read position cut off 63 and 188
read.df<-subset(sum.df,Read_pos>62 & Read_pos<188)
m.roc.df<-miseq.roc(read.df,possible_tp,possible_vars,">")

miseq.roc.p<-ggplot(m.roc.df,aes(x=1-adj.specificity,y=adj.sensitivity))+geom_step(aes(color=exp.freq),size=2)+scale_color_manual(values=rhg_cols)+xlab("1-Specificity")+ylab("Sensitivity")+guides(color=guide_legend(title="Frequency"))+scale_y_continuous(limits=c(0,1))#+scale_x_continuous(limits=c(0,0.005)

miseq.roc.p

# pdf("../results/figures/miseq_roc.one.sided.pdf")
# miseq.roc
# dev.off()

m.roc.table<-miseq.roc.table(read.df,0.01,possible_tp,possible_vars,">")
knitr::kable(m.roc.table)

ggplot(subset(sum.df,p.val<0.01),aes(x=Phred,y=MapQ,color=category))+geom_point(alpha=1)



```

### Figure 3

```{r,hiseq_roc,fig.width=3.4,fig.height=7}
#rm(list = setdiff(ls(), lsf.str())) # remove all variables except the functions
#tt1<-ttheme_minimal(colhead=list(fg_params=list(fontface=1,cex=0.75,fontfamily="serif")),core=list(fg_params=list(cex=0.75,fontfamily="serif")))


sum.df<-read.csv("../data/process/2015-6-23/Variants/all.sum.csv",stringsAsFactors = F,comment.char = "#") # reading in the variants from the pipeline output
mfe.df<-read.csv("../data/reference/mfe.csv") 
# Getting genome lenght and segment information from the wsn33 fasta file
reference.fasta<-"../data/reference/wsn33_wt_plasmid.fa" 
segments <- fasta.seqlengths(reference.fasta)
regions.bed <- data.frame(chr = gsub("[ ].*","", names(segments)), start=12, stop=segments-13, row.names=NULL) # the univeral primers are 12 and 13 bp long
regions.bed<-mutate(regions.bed,chr=as.character(chr))
# The samples are named in the following format log(genome copy/ul)_expected frequency of variants We'll use that information here to get the gc# and exp.freq for each variant called

pat<-"([0-9]+)_([0-9]+)" 
mutate(sum.df, gc = sub(pat,"\\1",Id),
       exp.freq = sub(pat,"\\2",Id)
)-> sum.df
mutate(sum.df,exp.freq=ifelse(test = grepl("0",exp.freq),yes=as.numeric(exp.freq)/1000,no = as.numeric(exp.freq)/100))->sum.df

sum.df<-subset(sum.df,exp.freq>0.001) # remove this to include exp.freq 0.01%
# Id the variants as T or F 
sum.df<-ddply(sum.df,~chr,primer.cut) # removes any variants outside of priming sites (in this case it removes 0)

true_snv<-read.csv("../data/reference/mutant_id.csv",stringsAsFactor=F) # get the T variants
sum.df<-mutate(sum.df,category=mutation %in% true_snv$mutant) # add Column for True and false variants
wt1_mut<-subset(sum.df,Id=="WT1" & freq.var>0.01,select=mutation)
wt2_mut<-subset(sum.df,Id=="WT2" &freq.var>0.01 ,select=mutation)

wt_mut<-intersect(wt1_mut$mutation,wt2_mut$mutation) # there aren't any of these

#length(wt_mut) How many wt muts do we find.
sum.df$category[sum.df$mutation %in% wt_mut]<-"wt"
sum.df<-subset(sum.df,!(grepl("WT",Id))) # remove the WT sample

sum.df$category[sum.df$mutation %in% mfe.df$mutation]<-"mfe"

possible_tp<-20 # there are 20 possible true positives in the mix
regions.bed<-mutate(regions.bed,length=stop-start-1) #because the numbers in the csv are the inner most binding sites of the primers and represent the last positiions that cannot be interogated.

total_possible_vars<-sum(regions.bed$length)*3 # three possible variants/ position

possible_vars<-total_possible_vars-length(wt_mut)-dim(mfe.df)[1]-20 # removing the mfe and wt_mut sites
hiseq_vars<-possible_vars
print(hiseq_vars)
hiseq_kb<-sum(regions.bed$length)/1000

h.roc.df<-hiseq.roc(sum.df,possible_tp,possible_vars,">")


x<-ddply(h.roc.df,~samp,fill_in_plots)

x<-mutate(x,frequency=find_freq.v.h(exp.freq))

x<-subset(x,gc==5)

x$adj.sensitivity[x$frequency=="1.0%"]<-x$adj.sensitivity[x$frequency=="1.0%"]-0.03
#x$adj.sensitivity[x$frequency=="5.0%"]<-x$adj.sensitivity[x$frequency=="5.0%"]+0.0

Hiseq.roc<-ggplot(subset(h.roc.df,gc==5),aes(x=1-adj.specificity,y=adj.sensitivity))+geom_step(aes(color=exp.freq),size=2)+scale_color_manual(values=rhg_cols,name="Frequency",labels=c("5.0%","2.0%","1.0%","0.5%","0.2%"))+xlab(bquote("1-Specificity (x"~10^-3~")"))+ylab("Sensitivity")+guides(color=guide_legend(title="Frequency"))+scale_y_continuous(limits=c(0,1))+scale_x_continuous(limits=c(0,0.005),breaks=c(0,0.001,0.002,0.003,0.004,0.005),labels=c(0:5))+theme(legend.position="none")+geom_text(data=subset(x,gc==5),aes(x=1-adj.specificity,y=adj.sensitivity,label=frequency),hjust=-0.2)

# This is so that the text is not cut off
Hiseq.roc <- ggplot_gtable(ggplot_build(Hiseq.roc))
Hiseq.roc$layout$clip[Hiseq.roc$layout$name == "panel"] <- "off"



h.table.df<-hiseq.roc.table(sum.df,0.01,possible_tp,possible_vars,">") 

#knitr::kable(subset(h.table.df,gc==5))

h.table.df<-rename(h.table.df,c("exp.freq"="Frequency","adj.sensitivity"="Sensitivity","TP"="True\nPositives","adj.specificity"="Specificity","FP"="False\nPositives"))

h.table.df<-subset(h.table.df,gc==5,select=-c(gc))
h.table.df$Frequency<-c("5.0%","2.0%","1.0%","0.5%","0.2%")
h.table.df$Sensitivity<-round(h.table.df$Sensitivity,digits=dd)
h.table.df$Specificity<-round(h.table.df$Specificity,digits=dd)



tbl<-make.table(h.table.df,tt1)    # Make table

readPNG("../results/pictures/hiseq_setup.png")->img # read in set up png
g <- rasterGrob(img, interpolate=TRUE)
qplot(1:1, 1:1, geom="blank") +
annotation_custom(g, xmin=-Inf, xmax=Inf, ymin=-Inf, ymax=Inf) +
geom_blank()+xlab("")+ylab("")+
theme(line = element_blank(),
text = element_blank(),
line = element_blank(),
title = element_blank())->hiseq.setup 

pdf("../results/figures/hiseq1.pdf",width=3.4,height=7)
ggdraw() +
  draw_plot(hiseq.setup, 0.1, .57, 0.8, 0.43) + #x y width height
  draw_plot(Hiseq.roc, 0.05, 0.33, .8, 0.33) +
  draw_plot(tbl, 0.05, 0, .9, 0.33) +
  draw_plot_label(c("A", "B", "C"), c(0.01, 0.01, 0.01), c(1, 0.7, 0.35), size = 15)
dev.off()

ggdraw() +
  draw_plot(hiseq.setup, 0.1, .57, 0.8, 0.43) + #x y width height
  draw_plot(Hiseq.roc, 0.05, 0.33, .8, 0.33) +
  draw_plot(tbl, 0.05, 0, .9, 0.33) +
  draw_plot_label(c("A", "B", "C"), c(0.01, 0.01, 0.01), c(1, 0.7, 0.35), size = 15)
# 
# knitr::kable(subset(h.table.df,gc==4))

twosided.hiseq<-mean(h.table.df$Specificity)
```


Figure 3. Decreased accuracy under patient-derived sample conditions A) 20 known single point mutants were diluted into WT WSN33 at decreasing frequencies. These populations were diluted further in basal media to match the genome concentrations found in patint-dervied samples (10$^5$-10$^3$ genomes/$\mu$l). B) The accuracy of DeepSNV in idenfying the 20 known true positive variants in the 10 $^5$ genomes/$\mu$l samples. C) A summary table of the accuracy of our method for the 10 $^5$ genomes/$\mu$l samples at a p value threshold of 0.01.

\pagebreak

### Figure 4 
```{r, phred_mapq filter}
sig.df<-subset(sum.df,p.val<0.01)

sig.df<-sig.df[order(sig.df$category,decreasing = F),]

mapq_phred.normal<-ggplot(subset(sig.df,category %in% c(T,F) & gc==5),aes(x=Phred,y=MapQ,color=category))+geom_point(size=0.8)+geom_hline(yintercept=20,linetype=2)+geom_vline(xintercept=30,linetype=2)+scale_x_continuous(limits=c(30,40))+scale_color_manual(name="",labels=c("False positives","True positives"),values=c("#b1bdc5ff","#08519cff"))+ylab("Mean MapQ")+xlab("Mean Phred ")+theme(legend.position="top") 

mapq<-ggplot(subset(sig.df,category %in% c(T,F) & gc==5),aes(x=MapQ,fill=category))+geom_histogram(position="dodge")+theme(legend.position="none")+scale_fill_manual(name="",labels=c("False positives","True positives"),values=c("#b1bdc5ff","#08519cff"))+ylab("SNV count")

phred<-ggplot(subset(sig.df,category %in% c(T,F) & gc==5),aes(x=Phred,fill=category))+geom_histogram(position="dodge")+theme(legend.position="none")+scale_fill_manual(name="",labels=c("False positives","True positives"),values=c("#b1bdc5ff","#08519cff"))+scale_x_continuous(limits = c(30,40),breaks = c(30,32.5,35,37.5,40))+ylab("SNV count")


pdf("../results/figures/mapq.pdf")
mapq
dev.off()

pdf("../results/figures/phred.pdf")
phred
dev.off()

#mapq_phred.normal

# pdf("~/Box Sync/Lab_work/Talks/2016_812/figures/mapq_phred.normal.pdf")
# mapq_phred.normal
# dev.off()

# mapq_phred.stingy<-ggplot(subset(sig.df,category %in% c(T,F) & gc==5),aes(x=Phred,y=MapQ,color=category))+geom_point(alpha=1)+scale_x_continuous(limits=c(30,40))+geom_hline(yintercept=30)+geom_vline(xintercept=35)+scale_color_hue(name="",labels=c("False positive","True positive"))+ylab("Mapping Quality")+xlab("Base Qualtiy (Phred) ")
# 
# mapq_phred.stingy
# 
# pdf("~/Box Sync/Lab_work/Talks/2016_812/figures/mapq_phred.stingy.pdf")
# mapq_phred.stingy
# dev.off()

#ggplot(subset(sig.df,gc==5 & category %in% c(T,F)),aes(x=Phred,y=MapQ,color=category))+geom_point()

#cut.df<-subset(sum.df,MapQ>30 & Phred>35) # This removes the variants that don't pass

cut.df.all<-sum.df
cut.df.all$p.val[cut.df.all$MapQ<30 | cut.df.all$Phred<35]<-1

# h.cut.df<-hiseq.roc(cut.df,possible_tp,possible_vars)
# 
# mapq_phred.roc<-ggplot(subset(h.cut.df,gc==5),aes(x=1-adj.specificity,y=adj.sensitivity))+geom_step(aes(color=exp.freq),size=2)+scale_color_brewer(palette = "Set1",name="Frequency",labels=c("5.0%","2.0%","1.0%","0.5%","0.2%"))+xlab(bquote("1-Specificity (x"~10^-3~")"))+ylab("Sensitivity")+guides(color=guide_legend(title="Frequency"))+scale_y_continuous(limits=c(0,1))+scale_x_continuous(limits=c(0,0.005),breaks=c(0,0.001,0.002,0.003,0.004,0.005),labels=c(0:5))
# 
# mapq_phred.roc
# 
# pdf("~/Box Sync/Lab_work/Talks/2016_812/figures/mapq_phred.rocpdf")
# mapq_phred.roc
# dev.off()
# 
# h.cut.table<-hiseq.roc.table(cut.df,0.01,possible_tp,possible_vars)
# 
# knitr::kable(subset(h.cut.table,gc==5))
```


```{r,read_pos,fig.width=3.4,fig.height=9}
# ggplot(sig.df,aes(x=pos,y=Read_pos,color=category))+geom_point()+facet_wrap(~chr)
# ggplot(subset(sig.df,gc==5),aes(x=pos,y=Read_pos,color=category))+geom_point()+facet_wrap(~chr)
# 
# ggplot(cut.df,aes(x=pos,y=Read_pos,color=category))+geom_point()+facet_wrap(~chr)
# ggplot(subset(cut.df,gc==5),aes(x=pos,y=Read_pos,color=category))+geom_point()+facet_wrap(~chr)

sig.cut.df<-subset(cut.df.all,p.val<0.01) 

#ggplot(sig.cut.df,aes(x=pos,y=Read_pos,color=category))+geom_point()+facet_wrap(~chr)
# ggplot(subset(sig.cut.df,gc==5),aes(x=pos,y=Read_pos,color=category))+geom_point()+facet_wrap(~chr)


read.pos<-ggplot(subset(sig.cut.df,category %in% c(T,F) & gc==5),aes(x=Read_pos,fill=category))+geom_histogram(position='dodge')+scale_fill_manual(name="",labels=c("False positive","True positive"),values=c("#b1bdc5ff","#08519cff"))+xlab("Average position on read")+ylab("Number of variants")+theme(legend.position="top")
# read.pos
# pdf("~/Box Sync/Lab_work/Talks/2016_812/figures/read.pos.pdf")
# read.pos
# dev.off()

fp_ends<-subset(sig.cut.df, category==F & Read_pos<32 | Read_pos>94)
false_p_phred<-mean(fp_ends$Phred)

#cut.read.df<-subset(cut.df,Read_pos>32 & Read_pos<94) # This removes the variants that don't pass

cut.read.df.all<-cut.df.all # this set's the p.val of variants that don't pass the cut to 1
cut.read.df.all$p.val[cut.read.df.all$Read_pos<32 | cut.read.df.all$Read_pos>94]<-1

h.cut.read.df<-hiseq.roc(cut.read.df.all,possible_tp,possible_vars,">")


x<-ddply(h.cut.read.df,~samp,fill_in_plots)

x<-mutate(x,frequency=find_freq.v.h(exp.freq))

x<-subset(x,gc==5)

x$adj.sensitivity[x$frequency=="1.0%"]<-x$adj.sensitivity[x$frequency=="1.0%"]-0.08
x$adj.sensitivity[x$frequency=="2.0%"]<-x$adj.sensitivity[x$frequency=="2.0%"]-0.01




read.roc<-ggplot(subset(h.cut.read.df,gc==5),aes(x=1-adj.specificity,y=adj.sensitivity))+geom_step(aes(color=exp.freq),size=2)+scale_color_manual(values=rhg_cols,name="Frequency",labels=c("5.0%","2.0%","1.0%","0.5%","0.2%"))+xlab(bquote("1-Specificity (x"~10^-3~")"))+ylab("Sensitivity")+guides(color=guide_legend(title="Frequency"))+scale_y_continuous(limits=c(0,1))+scale_x_continuous(limits=c(0,0.005),breaks=c(0,0.001,0.002,0.003,0.004,0.005),labels=c(0:5))+theme(legend.position="none")+geom_text(data=x,aes(x=1-adj.specificity,y=adj.sensitivity,label=frequency),hjust=-0.2)

# This is so that the text is not cut off
read.roc <- ggplot_gtable(ggplot_build(read.roc))
read.roc$layout$clip[read.roc$layout$name == "panel"] <- "off"


h.cut.read.table<-hiseq.roc.table(cut.read.df.all,0.01,possible_tp,possible_vars,">")
#knitr::kable(subset(h.cut.read.table,gc==5))

h.cut.read.table.5<-subset(h.cut.read.table,gc==5,select=-c(gc))
rename(h.cut.read.table.5,c("exp.freq"="Frequency","adj.sensitivity"="Sensitivity","TP"="True\nPositives","adj.specificity"="Specificity","FP"="False\nPositives"))->h.cut.read.table.5
h.cut.read.table.5$Frequency<-c("5.0%","2.0%","1.0%","0.5%","0.2%")
h.cut.read.table.5$Sensitivity<-round(h.cut.read.table.5$Sensitivity,digits=dd)
h.cut.read.table.5$Specificity<-round(h.cut.read.table.5$Specificity,digits=dd)

h.cut.read.table.5$Specificity[h.cut.read.table.5$Specificity>0.9999]<-">0.9999"

tbl<-make.table(h.cut.read.table.5,tt1) # Make table

pdf("../results/figures/figure4.pdf",width=3.4,height=9)
ggdraw() +
  draw_plot(mapq_phred.normal,0.03, 0.77, .85, 0.23) +
  draw_plot(read.pos, 0.03, .53, .85, .25) +
  draw_plot(read.roc, 0.03, 0.23, 0.85, .3) +
  draw_plot(tbl,   0.025, 0, 0.95, .25) +
  draw_plot_label(c("A", "B", "C","D"), c(0, 0, 0,0), c(1, 0.78, 0.55,0.25), size = 15)
dev.off()

ggdraw() +
  draw_plot(mapq_phred.normal,0.03, 0.77, .85, 0.23) + 
  draw_plot(read.pos, 0.03, .53, .85, .25) +
  draw_plot(read.roc, 0.03, 0.23, 0.85, .3) +
  draw_plot(tbl,   0.025, 0, 0.95, .25) +
  draw_plot_label(c("A", "B", "C","D"), c(0, 0, 0,0), c(1, 0.78, 0.55,0.25), size = 15)
```

Figure 4. Accuracy can be improved through quality thresholds. A) We stratified all called variants from the 10 $^5$ genomes/$\mu$l samples with p<0.01 stratified by the mean mapping quality of the reads containing the variant and the mean Phred scores of the variant bases. Dashed lines indicate  common cutoffs of 20 and 30 for mapping quality and Phred respectively. B) A histogram of average position on a pair end read the variants that passed our mean MapQ threshold of 30 and mean Phred threshold of 35. C) The accuracy of our analyis after applying the following quality cut offs : mean MapQ> 30, mean Phred> 35, and an average read position between 32 and 94 (the middle 50% of the read). D) A table summarizing accuracy in C at a p value threshold of 0.01.

\pagebreak

### Figure 5

```{r, frequency_accuracy,fig.width=3.4,fig.height=3.4}
# Here we'll just be using 

## Frequency
cut.read.sig.df<-subset(cut.read.df.all,p.val<0.01)
medians<-ddply(subset(cut.read.sig.df,category==T ),~exp.freq+gc, summarize, median=median(freq.var))

# ac_freq<-ggplot(subset(sum.df,category==T),aes(x=exp.freq,y=freq.var,fill=as.factor(exp.freq)))+facet_wrap(~gc)

#cut.read.sig.df$exp.freq<-as.factor(cut.read.sig.df$exp.freq)
#cut.read.sig.df$exp.freq <- factor(cut.read.sig.df$exp.freq, levels = rev(levels(cut.read.sig.df$exp.freq)))

### here weget the R2 of the sample

model<-lm(formula = freq.var~exp.freq,data = subset(cut.read.sig.df,category==T & gc=="5"))

corr_coef<-summary(model)$adj.r.squared


interest<-subset(cut.read.sig.df,category==T & gc=="5")

mean_fold<-mean(abs(interest$freq.var-interest$exp.freq)/interest$exp.freq)

print(mean_fold)

ac_freq<-ggplot(subset(cut.read.sig.df,category==T & gc=="5"),aes(x=exp.freq,y=freq.var,fill=factor(exp.freq,levels=rev(levels(as.factor(exp.freq))))))

ac_freq<-ac_freq+geom_dotplot(binaxis = "y", stackdir = "center",dotsize=1)+scale_fill_manual(values=rhg_cols)

ac_freq<-ac_freq+scale_y_log10(limits=c(0.001,0.1),breaks=c(0.001,0.002,0.005,0.01,0.02,0.05,0.1))#+theme(axis.text.y = element_text(colour = c("red","chartreuse4","black","chocolate2","azure4","blue","black"),face="bold"))
ac_freq<-ac_freq+scale_x_log10(limits=c(0.001,0.1),breaks=c(0.001,0.002,0.005,0.01,0.02,0.050,0.1))#+theme(axis.text.x = element_text(colour = c("red","chartreuse4","black","chocolate2","azure4","blue","black"),face="bold"))
ac_freq<-ac_freq+geom_abline(intercept = 0, slope = 1,linetype=2,size=1)
ac_freq<-ac_freq+xlab("Expected Frequency")+ylab("Measured Frequency")+theme(legend.position="none",axis.text.x = element_text(angle = 45, hjust = 1))

ac_freq<-ac_freq+geom_point(data=subset(medians,gc=="5"),mapping=aes(x=exp.freq,y=median),shape=95,size=20)
ac_freq

pdf("../results/figures/freq.pdf",width=3.4,height=3.4)
ac_freq
dev.off()

```

Figure 5. The accuracy of the frequency measurements for the known variants in the 10 $^5$ genomes/$\mu$l samples. The median of each distribution is shown as a black bar.


\pagebreak

### Figure 6

```{r,lofreq,fig.height=5,fig.width=3.4}

lofreq.df<-read.csv("../data/process/2015-6-23/lofreq/all.removed.mapq_vcf.vcf_csv.csv",stringsAsFactors = F)

lofreq.df$category[lofreq.df$mutation %in% wt_mut]<-"wt"
lofreq.df$categroy[lofreq.df$mutation %in% mfe.df$mutation]<-"mfe"

lofreq.df$p.val<-lofreq.df$QUAL # to make the roc based on the quality scores given by lofreq

lofreq.roc<-hiseq.roc(lofreq.df,possible_tp,possible_vars,"<")

x<-ddply(lofreq.roc,~samp,fill_in_plots)


x<-mutate(x,frequency=as.character(find_freq.v.h(exp.freq)))

x<-subset(x,gc==5)

x$adj.sensitivity[x$frequency=="1.0%"]<-x$adj.sensitivity[x$frequency=="1.0%"]+0.02
x$adj.sensitivity[x$frequency=="2.0%"]<-x$adj.sensitivity[x$frequency=="2.0%"]-0.02



ggplot(subset(lofreq.roc,gc==5),aes(x=1-adj.specificity,y=adj.sensitivity))+geom_step(aes(color=exp.freq),size=2)+scale_color_manual(values=rhg_cols)+xlab("1-Specificity")+ylab("Sensitivity")+guides(color=guide_legend(title="Frequency"))+scale_y_continuous(limits=c(0,1))+scale_x_continuous(limits=c(0,0.005),breaks=c(0,0.001,0.002,0.003,0.004,0.005),labels=c(0:5))+theme(legend.position='none')+geom_text(data=x,aes(x=1-adj.specificity,y=adj.sensitivity,label=frequency),hjust=-0.2)->lofreq.roc.p
# This is so that the text is not cut off
lofreq.roc.p <- ggplot_gtable(ggplot_build(lofreq.roc.p))
lofreq.roc.p$layout$clip[lofreq.roc.p$layout$name == "panel"] <- "off"


#lofreq claims no further monkey bussiness is needed to we'll just make a table with all its output
ddply(lofreq.df,~Id,summarize,gc=gc[1],exp.freq=exp.freq[1],TP=length(which(category==T)),Sensitivity=TP/20,FP=length(which(category==F)),Specificity=(possible_vars-possible_tp-FP)/(possible_vars-possible_tp))->lofreq.table
lofreq.table<-lofreq.table[order(lofreq.table$exp.freq,decreasing = T),]

#knitr::kable(subset(lofreq.table,gc==5,select=c(exp.freq,TP,FP)))
lofreq.table<-head(subset(lofreq.table,gc==5,select=-c(gc,Id)),n=-1) # don't need the 0.1% sample
rename(lofreq.table,c("exp.freq"="Frequency","TP"="True\nPositives","FP"="False\nPositives"))->lofreq.table
lofreq.table$Frequency<-c("5.0%","2.0%","1.0%","0.5%","0.2%")
lofreq.table$Sensitivity<-round(lofreq.table$Sensitivity,digits=dd)
lofreq.table$Specificity<-round(lofreq.table$Specificity,digits=dd)
lofreq.table<-subset(lofreq.table,select=c("Frequency","Sensitivity","True\nPositives","Specificity","False\nPositives"))
tt2<-ttheme_minimal(colhead=list(fg_params=list(fontface=1,cex=0.65,fontfamily="serif")),core=list(fg_params=list(cex=0.65,fontfamily="serif")))

tbl<-make.table(lofreq.table,tt1) # Make table

# ggdraw() +
#   draw_plot(lofreq.roc.p, 0, .5, 0.5, .5) +
#   draw_plot(tbl, 0.5, 0.5, 0.5, .5) +
#   draw_plot_label(c("A", "B"), c(0, 0.5), c(1, 1), size = 15)


pdf("../results/figures/lofreq1.pdf",width=3.4,height=5)
ggdraw() +
  draw_plot(lofreq.roc.p, 0, 0.5, 0.9, 0.45) +
  draw_plot(tbl, 0.025, 0, 0.95, .48) +
  draw_plot_label(c("A", "B"), c(0, 0), c(1, 0.5), size = 15)
dev.off()

ggdraw() +
  draw_plot(lofreq.roc.p, 0, 0.5, 0.9, 0.45) + 
  draw_plot(tbl, 0.025, 0, 0.95, .48) +
  draw_plot_label(c("A", "B"), c(0, 0), c(1, 0.5), size = 15)
```

Figure 6. The accuracy of the Lofreq algorithm on our 10^5^ genomes/$\mu$l populations. A)  The accuracy of Lofreq using standard parameters. The specificity of the caller was scaled to account for the same number of tests as the DeepSNV algorithm. B) A table summarizing the accuracy of the ROC A using the standard cut offs applied by the algorithm.

\pagebreak


### Figure 7 

```{r,roc_lower_inputs4,fig.width=3.4,fig.height=9}

x<-ddply(h.cut.read.df,~samp,fill_in_plots)
x<-subset(x,gc==4)

x<-mutate(x,frequency=find_freq.v.h(exp.freq))

x$adj.sensitivity[x$frequency=="1.0%"]<-x$adj.sensitivity[x$frequency=="1.0%"]-0.05
x$adj.sensitivity[x$frequency=="2.0%"]<-x$adj.sensitivity[x$frequency=="2.0%"]+0.05
x$adj.sensitivity[x$frequency=="0.5%"]<-x$adj.sensitivity[x$frequency=="0.5%"]+0.025



roc.4<-ggplot(subset(h.cut.read.df,gc==4),aes(x=1-adj.specificity,y=adj.sensitivity))+geom_step(aes(color=exp.freq),size=2)+scale_color_manual(values=rhg_cols,name="Frequency",labels=c("5.0%","2.0%","1.0%","0.5%","0.2%"))+xlab(bquote("1-Specificity (x"~10^-3~")"))+ylab("Sensitivity")+guides(color=guide_legend(title="Frequency"))+scale_y_continuous(limits=c(0,1))+scale_x_continuous(limits=c(0,0.005),breaks=c(0,0.001,0.002,0.003,0.004,0.005),labels=c(0:5))+theme(legend.position="none")+geom_text(data=x,aes(x=1-adj.specificity,y=adj.sensitivity,label=frequency),hjust=-0.2)


# This is so that the text is not cut off
roc.4 <- ggplot_gtable(ggplot_build(roc.4))
roc.4$layout$clip[roc.4$layout$name == "panel"] <- "off"


# roc.4
# 
# pdf("~/Box Sync/Lab_work/Talks/2016_812/figures/roc.4.pdf")
# roc.4
# dev.off()
# 
# knitr::kable(subset(h.cut.read.table,gc==4))

x<-ddply(h.cut.read.df,~samp,fill_in_plots)
x<-subset(x,gc==3)

x<-mutate(x,frequency=find_freq.v.h(exp.freq))

roc.3<-ggplot(subset(h.cut.read.df,gc==3),aes(x=1-adj.specificity,y=adj.sensitivity))+geom_step(aes(color=exp.freq),size=2)+scale_color_manual(values=rhg_cols,name="Frequency",labels=c("5.0%","2.0%","1.0%","0.5%","0.2%"))+xlab(bquote("1-Specificity (x"~10^-3~")"))+ylab("Sensitivity")+guides(color=guide_legend(title="Frequency"))+scale_y_continuous(limits=c(0,1))+scale_x_continuous(limits=c(0,0.005),breaks=c(0,0.001,0.002,0.003,0.004,0.005),labels=c(0:5))+theme(legend.position="none")+geom_text(data=x,aes(x=1-adj.specificity,y=adj.sensitivity,label=frequency),hjust=-0.2)

# This is so that the text is not cut off
roc.3 <- ggplot_gtable(ggplot_build(roc.3))
roc.3$layout$clip[roc.3$layout$name == "panel"] <- "off"


# pdf("~/Box Sync/Lab_work/Talks/2016_812/figures/roc.3.pdf")
# roc.3
# dev.off()
# 
# 
# knitr::kable(subset(h.cut.read.table,gc==3))


h.cut.read.table.4<-subset(h.cut.read.table,gc==4,select=-c(gc))
rename(h.cut.read.table.4,c("exp.freq"="Frequency","adj.sensitivity"="Sensitivity","TP"="True\nPositives","adj.specificity"="Specificity","FP"="False\nPositives"))->h.cut.read.table.4
h.cut.read.table.4$Frequency<-c("5.0%","2.0%","1.0%","0.5%","0.2%")
h.cut.read.table.4$Sensitivity<-round(h.cut.read.table.4$Sensitivity,digits=dd)
h.cut.read.table.4$Specificity<-round(h.cut.read.table.4$Specificity,digits=dd)

tbl.4<-make.table(h.cut.read.table.4,tt1) # Make table


h.cut.read.table.3<-subset(h.cut.read.table,gc==3,select=-c(gc))
rename(h.cut.read.table.3,c("exp.freq"="Frequency","adj.sensitivity"="Sensitivity","TP"="True\nPositives","adj.specificity"="Specificity","FP"="False\nPositives"))->h.cut.read.table.3
h.cut.read.table.3$Frequency<-c("5.0%","2.0%","1.0%")
h.cut.read.table.3$Sensitivity<-round(h.cut.read.table.3$Sensitivity,digits=dd)
h.cut.read.table.3$Specificity<-round(h.cut.read.table.3$Specificity,digits=dd)

tbl.3<-make.table(h.cut.read.table.3,tt1) # Make table

pdf("../results/figures/lower.pdf",width=3.4,height=9)
ggdraw() +
  draw_plot(roc.4,0.03, 0.69, .85, 0.25) +
  draw_plot(tbl.4, 0.025, .47,  0.95, .25) +
  draw_plot(roc.3, 0.03, 0.22, 0.85, .25) +
  draw_plot(tbl.3,   0.025, 0, 0.95, .25) +
  draw_plot_label(c("A", "B", "C","D"), c(0, 0, 0,0), c(0.97, 0.73, 0.5,0.23), size = 15)
dev.off()

ggdraw() +
  draw_plot(roc.4,0.03, 0.69, .85, 0.25) + 
  draw_plot(tbl.4, 0.025, .47,  0.95, .25) +
  draw_plot(roc.3, 0.03, 0.22, 0.85, .25) +
  draw_plot(tbl.3,   0.025, 0, 0.95, .25) +
  draw_plot_label(c("A", "B", "C","D"), c(0, 0, 0,0), c(0.97, 0.73, 0.5,0.23), size = 15)
```

Figure 7. The accuracy of DeepSNV decreased when applied to samples with less input nucleic acid. A) An ROC curve for the 10 $^4$ genomes/$\mu$l samples. B) A table summarizing the accuracy at 10 $^4$ genomes/$\mu$l with a  p value threshold of 0.01. C) An ROC curve depicting the accuracy of our method when applied to the 10 $^3$ genomes/$\mu$l samples. B) A table summarizing the accuracy at 10 $^3$ genomes/$\mu$l with a  p value threshold of 0.01

\pagebreak

### Figure 8

```{r,duplicates,fig.height=5,fig.width=3.4}
dup.df<-read.csv("../data/process/2015-11-14/Variants/all.sum.csv",stringsAsFactors = F,comment.char = "#")
#sum.df<-read.csv("~/Box Sync/Lab_work/NGS/Benchmarking/data/2015-11-14/all.sum.csv")
pat<-"([0-9]+)_([0-9]+)_([0-9]+)"
mutate(dup.df, exp.freq = sub(pat,"\\2",Id),
       dup = as.numeric(sub(pat,"\\3",Id)),
)-> dup.df

#dup.df<-subset(dup.df,Id!="PC") # remove the plasmid control that was run
mutate(dup.df,Id=paste(4,exp.freq,sep="_"),exp.freq=ifelse(test = grepl("0",exp.freq),yes=as.numeric(exp.freq)/1000,no = as.numeric(exp.freq)/100))->dup.df


dup.df<-mutate(dup.df,category=mutation %in% true_snv$mutant) # add Column for True and false variants
dup.df$category[dup.df$mutation %in% wt_mut]<-"wt"

cut.df<-subset(dup.df,Phred>35 & MapQ>30 & Read_pos<94 & Read_pos>32)

dups<-ddply(cut.df,~exp.freq+mutation,summarize,freq1=freq.var[1],freq2=freq.var[2],p.val1=p.val[1],p.val2=p.val[2],category=category[1],Id=Id[1])
dups$freq2[is.na(dups$freq2)]<-0
dups$p.val2[is.na(dups$p.val2)]<-1

dups<-mutate(dups,p.val=pmax(p.val1,p.val2))

#dups<-subset(dups,p.val!=1)

dups.roc<-hiseq.roc(dups,possible_tp,possible_vars,">")


x<-ddply(dups.roc,~samp,fill_in_plots)


x<-mutate(x,frequency=find_freq.v.h(exp.freq))

x$adj.sensitivity[x$frequency=="1.0%"]<-x$adj.sensitivity[x$frequency=="1.0%"]-0.18
x$adj.sensitivity[x$frequency=="2.0%"]<-x$adj.sensitivity[x$frequency=="2.0%"]-0.09


dups.roc.p<-ggplot(dups.roc,aes(x=1-adj.specificity,y=adj.sensitivity))+geom_step(aes(color=exp.freq),size=2)+scale_color_manual(values=rhg_cols,name="Frequency",labels=c("5.0%","2.0%","1.0%","0.5%","0.2%"))+xlab(bquote("1-Specificity (x"~10^-3~")"))+ylab("Sensitivity")+guides(color=guide_legend(title="Frequency"))+scale_y_continuous(limits=c(0,1))+scale_x_continuous(limits=c(0,0.005),breaks=c(0,0.001,0.002,0.003,0.004,0.005),labels=c(0:5))+theme(legend.position="none")+geom_text(data=x,aes(x=1-adj.specificity,y=adj.sensitivity,label=frequency),hjust=-0.2)

# This is so that the text is not cut off
dups.roc.p <- ggplot_gtable(ggplot_build(dups.roc.p))
dups.roc.p$layout$clip[dups.roc.p$layout$name == "panel"] <- "off"


# dups.roc.p
# pdf("~/Box Sync/Lab_work/Talks/2016_812/figures/dups.roc.pdf")
# dups.roc.p
# dev.off()

dups.table<-hiseq.roc.table(dups,0.01,possible_tp,possible_vars,">")
#knitr::kable(subset(dups.table))

rename(dups.table,c("exp.freq"="Frequency","adj.sensitivity"="Sensitivity","TP"="True\nPositives","adj.specificity"="Specificity","FP"="False\nPositives"))->dups.table
subset(dups.table,select=-c(gc))->dups.table
dups.table$Frequency<-c("5.0%","2.0%","1.0%","0.5%")
dups.table$Sensitivity<-round(dups.table$Sensitivity,digits=dd)
dups.table$Specificity<-round(dups.table$Specificity,digits=dd)

tbl<-make.table(dups.table,tt1) # Make table

exp.meas.freq<-ddply(cut.df,~exp.freq+mutation+category, function(x){
                                                                                      freq1=x$freq.var[x$dup=="1"]
                                                                                      freq2=x$freq.var[x$dup=="2"]
                                                                                      if(length(freq1)==0) freq1=0
                                                                                      if(length(freq2)==0) freq2=0
                                                                                      out=data.frame(freq1=freq1,freq2=freq2)})



pdf("../results/figures/dups.roc.pdf",width=3.4,height=5)
ggdraw() +
  draw_plot(dups.roc.p, 0, 0.5, 0.9, 0.45) +
  draw_plot(tbl, 0.025, 0, 0.95, .48) +
  draw_plot_label(c("A", "B"), c(0, 0), c(1, 0.5), size = 15)
dev.off()

ggdraw() +
  draw_plot(dups.roc.p, 0, 0.5, 0.9, 0.45) + 
  draw_plot(tbl, 0.025, 0, 0.95, .48) +
  draw_plot_label(c("A", "B"), c(0, 0), c(1, 0.5), size = 15)









# ggplot(subset(exp.meas.freq),aes(x=freq1,y=freq2,color=category))+geom_point(aes(size=3))+scale_y_log10()+scale_x_log10()+xlab("Frequency in sample 1")+ylab("Frequency in sample 2")+scale_color_hue(name="",labels=c("False positive","True positive"))+theme(legend.position=c(0.5,0.95))+geom_abline(intercept=0,slope=1,lty=2)->dup.freq
# dup.freq
# 
# pdf("~/Box Sync/Lab_work/Talks/2016_812/figures/dup.freq.pdf")
# dup.freq
# dev.off()

```

Figure 8. The accuracy of DeepSNV at lower input levels is improved by processing samples in duplicate. A) An ROC curve of  the 10 $^4$ genomes/$\mu$l samples processed in duplicate. A variant was required to be found in both duplicates to be considered. B) A table summarizing the accuracy when applied to duplicate samples with 10 $^4$ genomes/$\mu$l input, p<0.01. 

\pagebreak

### Table 1.

Table 1. Accurate SNV indentification is vital for accurate measure of intrahost diversity. We measured the richness and Shannon's Entropy of the designated 10^5^ genomes$\mu$l populations at each stage of our benchmarking procedure and compared the results with what is expected. Additionaly we calculated the L1-norm of the populations relative to the expected population. 

```{r,results="asis",cache=FALSE}
mut_pat="(.*)_[A,T,C,G](\\d+)[A,T,C,G]"
perfect<-data.frame(mutation=rep(true_snv$mutant,times=4),freq.var=rep(c(0.05,0.02,0.01,0.005),each=20))
perfect=mutate(perfect,chr=gsub(pattern = mut_pat,replacement = "\\1",x=mutation),pos=gsub(pattern = mut_pat,replacement = "\\2",x=mutation))
dups<-subset(dups,p.val<0.01)
dups<-mutate(dups,freq.var=pmax(freq1,freq2))

rich<-function(x){
  dim(x)[1]  
}

H<-function(x){
  H_pos<-ddply(x,~chr+pos,summarize,wt=1-sum(freq.var),H=-sum(c(freq.var*log(freq.var),wt*log(wt))))
sum(H_pos$H)/(possible_vars/3)
}

D<-function(x){
  D_pos<-ddply(x,~chr+pos,summarize,wt=1-sum(freq.var),D=1/sum(c(freq.var^2,wt^2)))
  others=(possible_vars/3)-dim(D_pos)[1]
  sum(D_pos$D,others)/(possible_vars/3)
}

euc<-function(x,ref){
  x$Id="test"
  ref$Id="ref"
  all<-rbind(subset(x,select=c(mutation,freq.var,Id)),subset(ref,select=c(mutation,freq.var,Id)))
  tab<-as.matrix(subset(dcast(all,mutation ~ Id, value.var = "freq.var"),select=-c(mutation)))
  tab[is.na(tab)]<-0
  dist(t(tab),"manhattan")[1]
}
get_data<-function(x,setting){
  if(length(unique(x$freq.var))==1){
    exp.freq=unique(x$freq.var) # this is the perfect condition where there is only 1 frequency and it is all the same
  }else{
  exp.freq=unique(x$exp.freq)
  }
  out<-data.frame(Richness=round(rich(x),digits = 0), Entropy=format(H(x),scientific = T,digits = 3),'L1-norm'=round(euc(x,subset(perfect,freq.var==exp.freq)),digits=3)) #"1/D"=D(x),
  rownames(out)<-setting
  return(t(out))
}
require(xtable)
tabulate<-function(input,freq){
x<-data.frame()
for(i in 1:length(freq)){  
get_data(subset(perfect,freq.var==freq[i]),"Expected")->perf
get_data(subset(sum.df,gc==input & exp.freq==freq[i] & p.val<0.01 & category %in% c(T,F)),"DeepSNV")->deepSNV
get_data(subset(lofreq.df,gc==input & exp.freq==freq[i] & category %in% c(T,F)),"Lofreq")->lo
get_data(subset(cut.read.df.all,gc==input & exp.freq==freq[i] & p.val<0.01 & category %in% c(T,F) ),"DeepSNV modified")->deepSNV.f
y<-cbind(perf,deepSNV.f,lo,deepSNV)
x<-rbind(x,y)
}
names<-data.frame(Frequency=c("","5.0%","","","1.0%","","","0.5%",""),names=rep(c("Richness","Entropy","L1-norm"),times=3))
x<-cbind(names,x)
#x<-subset(x,select=c(names,Expected,DeepSNV,Lofreq,DeepSNV Final))
x<-rename(x,c("names"=""))
write.csv(x,file="../results/tables/table_1.csv")
xtab<-xtable(x)
align(xtab) <- rep("c", 7)
print(xtab,include.rownames = FALSE,hline.after=c(-1,0,3,6))

}


#

```


```{r,results="asis",cache=F}
tabulate(5,c(0.05,0.01,0.005))
#

```


```{r,coverage}
cov<-mutate(subset(cut.read.df.all,p.val<0.01 & gc>2 & exp.freq>0.001),coverage=cov.tst.fw+cov.tst.bw)

tp<-subset(cov,category==T)
fp<-subset(cov,category==F)
tp_cov<-data.frame(coverage=min(tp$coverage),freq.var=tp$freq.var[tp$coverage==min(tp$coverage)])
fp_cov<-data.frame(coverage=min(fp$coverage),freq.var=fp$freq.var[fp$coverage==min(fp$coverage)])
```



```{r,oneside.Hiseq}
sum.df<-read.csv("../data/process/2015-6-23.one.sided/Variants/all.sum.csv",stringsAsFactors = F,comment.char = "#") # reading in the variants from the pipeline output
mfe.df<-read.csv("../data/reference/mfe.csv")
# Getting genome lenght and segment information from the wsn33 fasta file
reference.fasta<-"../data/reference/wsn33_wt_plasmid.fa"
segments <- fasta.seqlengths(reference.fasta)
regions.bed <- data.frame(chr = gsub("[ ].*","", names(segments)), start=12, stop=segments-13, row.names=NULL) # the univeral primers are 12 and 13 bp long
regions.bed<-mutate(regions.bed,chr=as.character(chr))

# The samples are named in the following format log(genome copy/ul)_expected frequency of variants We'll use that information here to get the gc# and exp.freq for each variant called

pat<-"([0-9]+)_([0-9]+)" 
mutate(sum.df, gc = sub(pat,"\\1",Id),
       exp.freq = sub(pat,"\\2",Id)
)-> sum.df
mutate(sum.df,exp.freq=ifelse(test = grepl("0",exp.freq),yes=as.numeric(exp.freq)/1000,no = as.numeric(exp.freq)/100))->sum.df

sum.df<-subset(sum.df,exp.freq>0.001) # remove this to include exp.freq 0.01%
# Id the variants as T or F 
true_snv<-read.csv("../data/reference/mutant_id.csv",stringsAsFactor=F) # get the T variants
sum.df<-mutate(sum.df,category=mutation %in% true_snv$mutant) # add Column for True and false variants
sum.df<-ddply(sum.df,~chr,primer.cut)

wt1_mut<-subset(sum.df,Id=="WT1" & freq.var>0.01,select=mutation)
wt2_mut<-subset(sum.df,Id=="WT2"& freq.var>0.01 ,select=mutation)

wt_mut<-intersect(wt1_mut$mutation,wt2_mut$mutation)

#length(wt_mut) How many wt muts do we find.
sum.df$category[sum.df$mutation %in% wt_mut]<-"wt"
sum.df<-subset(sum.df,!(grepl("WT",Id))) # remove the WT sample

sum.df$category[sum.df$mutation %in% mfe.df$mutation]<-"mfe"

possible_tp<-20 # there are 20 possible true positives in the mix
regions.bed<-mutate(regions.bed,length=stop-start-1) #because the numbers in the csv are the inner most binding sites of the primers and represent the last positiions that cannot be interogated.

total_possible_vars<-sum(regions.bed$length)*3 # three possible variants/ position

possible_vars<-total_possible_vars-length(wt_mut)-dim(mfe.df)[1] # removing the mfe and wt_mut sites
hiseq_vars<-possible_vars
hiseq_kb<-sum(regions.bed$length)/1000

h.roc.df<-hiseq.roc(sum.df,possible_tp,possible_vars,">")


x<-ddply(h.roc.df,~samp,fill_in_plots)

x<-mutate(x,frequency=find_freq.v.h(exp.freq))

x<-subset(x,gc==5)

x$adj.sensitivity[x$frequency=="1.0%"]<-x$adj.sensitivity[x$frequency=="1.0%"]-0.03
#x$adj.sensitivity[x$frequency=="5.0%"]<-x$adj.sensitivity[x$frequency=="5.0%"]+0.0

Hiseq.roc<-ggplot(subset(h.roc.df,gc==5),aes(x=1-adj.specificity,y=adj.sensitivity))+geom_step(aes(color=exp.freq),size=2)+scale_color_manual(values=rhg_cols,name="Frequency",labels=c("5.0%","2.0%","1.0%","0.5%","0.2%"))+xlab(bquote("1-Specificity (x"~10^-3~")"))+ylab("Sensitivity")+guides(color=guide_legend(title="Frequency"))+scale_y_continuous(limits=c(0,1))+scale_x_continuous(limits=c(0,0.005),breaks=c(0,0.001,0.002,0.003,0.004,0.005),labels=c(0:5))+theme(legend.position="none")+geom_text(data=subset(x,gc==5),aes(x=1-adj.specificity,y=adj.sensitivity,label=frequency),hjust=-0.2)

# This is so that the text is not cut off
Hiseq.roc <- ggplot_gtable(ggplot_build(Hiseq.roc))
Hiseq.roc$layout$clip[Hiseq.roc$layout$name == "panel"] <- "off"



h.table.df<-hiseq.roc.table(sum.df,0.01,possible_tp,possible_vars,">") 

#knitr::kable(subset(h.table.df,gc==5))

h.table.df<-rename(h.table.df,c("exp.freq"="Frequency","adj.sensitivity"="Sensitivity","TP"="True\nPositives","adj.specificity"="Specificity","FP"="False\nPositives"))

h.table.df<-subset(h.table.df,gc==5,select=-c(gc))
h.table.df$Frequency<-c("5.0%","2.0%","1.0%","0.5%","0.2%")
h.table.df$Sensitivity<-round(h.table.df$Sensitivity,digits=dd)
h.table.df$Specificity<-round(h.table.df$Specificity,digits=dd)



tbl<-make.table(h.table.df,tt1)    # Make table

readPNG("../results/pictures/hiseq_setup.png")->img # read in set up png
g <- rasterGrob(img, interpolate=TRUE)
qplot(1:1, 1:1, geom="blank") +
annotation_custom(g, xmin=-Inf, xmax=Inf, ymin=-Inf, ymax=Inf) +
geom_blank()+xlab("")+ylab("")+
theme(line = element_blank(),
text = element_blank(),
line = element_blank(),
title = element_blank())->hiseq.setup 

# ggdraw() +
#   draw_plot(hiseq.setup, 0.1, .57, 0.8, 0.43) + #x y width height
#   draw_plot(Hiseq.roc, 0.05, 0.33, .8, 0.33) +
#   draw_plot(tbl, 0.05, 0, .9, 0.33) +
#   draw_plot_label(c("A", "B", "C"), c(0.01, 0.01, 0.01), c(1, 0.7, 0.35), size = 15)
# # 
# knitr::kable(subset(h.table.df,gc==4))

onesided.hiseq<-mean(h.table.df$Specificity)
onesided.min<-min(h.table.df$Specificity)

```


```{r,table2}
data<-read.csv("../data/reference/patient.csv")
data<-subset(data,p.val<0.01 & freq.var<0.9)
data<-ddply(data,~chr,primer.cut) # removes sites out side of primers

data.cut<-subset(data,Phred>35 & MapQ>30 & Read_pos<94 & Read_pos>32)

possible_vars<-total_possible_vars
get_data<-function(x){
  out<-data.frame(Richness=round(rich(x),digits = 0), Entropy=format(H(x),scientific = T,digits = 3)) #"1/D"=D(x),
  #rownames(out)<-setting
  return(out)
}
get_data.cut<-function(x){
  out<-data.frame(Richness.mod=round(rich(x),digits = 0), Entropy.mod=format(H(x),scientific = T,digits = 3)) #"1/D"=D(x),
  #rownames(out)<-setting
  return(out)
}

x<-ddply(data,~Id+DPI+FINAL_RESULT+season+log10(gc_ul),get_data)

y<-ddply(data.cut,~Id+DPI+log10(gc_ul),get_data.cut)

x<-join(x,y)
x<-subset(x,!(Id%in% c(1240,1241)))
x<-x[order(x$`log10(gc_ul)`,decreasing = T),]
write.csv(x,"../results/tables/table_2.cali_vic.csv")
ggplot(data,aes(pos,freq.var))+geom_point()+facet_wrap(~chr)

ggplot(data.cut,aes(pos,freq.var))+geom_point()+facet_wrap(~chr)

ggplot(data,aes(freq.var))+geom_histogram()+scale_x_log10()

ggplot(data.cut,aes(freq.var))+geom_histogram()+scale_x_log10()+scale_y_log10()

x<-subset(data.cut,Id %in% c(1376,1401,1405,1374,1227,1321,1229,1245))
```
